snapshot-labs/sx-monorepo
# feat: add proposals timeline for followed spaces

### Summary



Closes: #217 
 
![Screenshot 2024-04-13 at 15 35 29](https://github.com/snapshot-labs/sx-monorepo/assets/495709/cb08d341-b988-4d9b-be78-6ea51a83cecd)

### Features

- New timeline page (route `/home`): show all the proposals an account is following from snapshot v1

### Refactoring/Changes

- `loadProposals` API has been changed to take an array of space IDs, instead of just a single ID
- `loadVotes` API has been changed to make the `spaceId` argument optional
- Rename the existing Home.vue (route `/`) to Landing.vue
- The new timeline page (/home) and existing explore page (/explore) now share a common sidebar, and are nested under the 'my' folder and route.
- The /home and /explore are now guarded, and available only for logged in users
- The ProposalsList component now takes an addition `show-space` props, to toggle display of space name alongside proposal title
- Add a new ref in useWeb composable to register whether the authentication process has started or not. This is in order to avoid a flashing message on page load, when the login process has not completed yet, because currently, everyone is guest on page load, until login process has finished. This ref will allow us to show the guest message only after login has finished.
- ~~The new `follows` list (list of spaceIds) is an optional property of `User`~~ Add a new Follow type and `loadFollows` to the API, anticipating #282 

### How to test

1. Go to http://localhost:8080/#/home as a logged in user
2. It should show you all the proposals from your followed spaces (you should have followed some spaces with the same account on v1)
3. By logging out, it should redirect you to the `/` landing page

### To-Do

- [x] Show space name ~~+ icon~~ alongside each proposals
- [x] Load follows list from correct network
- [x] Fix "no proposals" message flashing on page load
- [x] Find a more appropriate route name
- [x] Add "Voted" indicator
- [x] Reset the proposals list when user is changing account
